XML.java

```diff
@@ -533,7 +533,7 @@ public static String toString(Object object, String tagName)
             while (keys.hasNext()) {
                 key = keys.next();
                 value = jo.opt(key);
-                if (value == null) {
+                if (value == null || JSONObject.NULL.equals(value)) {
```

Comment:
you don't need the OR here, simply `if (JSONObject.NULL.equals(value)) {` will do the same thing